### PR TITLE
[Documentation][Cookbook] How to add images to an entity?

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q               | A
| --------------- | ---
| Doc fix?        | no |
| New docs?    | yes :tada: |
| Related tickets |  |
| License         | MIT |

### CODE REVIEW

File: docs/cookbook/images-on-entity.rst [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+    sylius_shipping:
+        resources:
+            shipping_method:
+                classes:
+                    model: AppBundle\Entity\ShippingMethod
+
+3. Create the ShippingMethodImage class
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In the ``AppBundle/Entity`` namespace place the ``ShippingMethodImage`` class which should look like that:
```
Comment (sweoggy) [inline docs/cookbook/images-on-entity.rst]: that -> this

File: docs/cookbook/images-on-entity.rst [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+        public function getBlockPrefix()
+        {
+            return 'app_shipping_method_image';
+        }
+    }
+
+8. Register the ShippingMethodImageType as a service
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+After creating the form type class, you need to register it as a ``form.type`` service liek below:
```
Comment (sweoggy) [inline docs/cookbook/images-on-entity.rst]: liek -> like

File: docs/cookbook/images-on-entity.rst [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+10. Extend the ShippingMethodType with the images field
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. tip::
+
+    Read more about :doc:`customizing forms via extensions in the dedicated guide </customization/form>`.
+
+**Create the form extension class** for the ``Sylius\Bundle\ShippingBundle\Form\Type\ShippingMethodType``:
+
+It needs to have the images field as a CollectionType. If you want to give a possibility to add more than one image to the entity
```
Comment (sweoggy) [inline docs/cookbook/images-on-entity.rst]: a possibility -> the possibility

File: docs/cookbook/images-on-entity.rst [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+                - { name: kernel.event_listener, event: "sylius.taxon.pre_update", method: "uploadImage" }
+                - { name: kernel.event_listener, event: "sylius.shipping_method.pre_create", method: "uploadImage" }
+                - { name: kernel.event_listener, event: "sylius.shipping_method.pre_update", method: "uploadImage" }
+
+12. Render the images field in the form view
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In order to achieve that you will need to customize the form view from the ``SyliusAdminBundle/views/ShippingMethod/_form.html.twig`` file.
+
+Copy and pase its contents into your own ``app/Resources/SyliusAdminBundle/views/ShippingMethod/_form.html.twig`` file,
```
Comment (venyii) [inline docs/cookbook/images-on-entity.rst]: pase -> paste

File: docs/cookbook/images-on-entity.rst [hunk 5]
```diff
@@ -0,0 +1,387 @@
...
+    services:
+        app.form.extension.type.shipping_method:
+            class: AppBundle\Form\Extension\ShippingMethodTypeExtension
+            tags:
+                - { name: form.type_extension, extended_type: Sylius\Bundle\ShippingBundle\Form\Type\ShippingMethodType }
+
+11. Override the definition of the ImageUploader service
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In order to handle the image upload you need to attach the image upload listener to the ShippingMethod entity events:
```
Comment (venyii) [inline docs/cookbook/images-on-entity.rst]:
```
``ShippingMethod``
```
With namespace? Not sure if necessary

**Comment by TheMadeleine** [comment]
@sweoggy, @venyii Thanks a lot for your review guys! :)

File: docs/cookbook/images-on-entity.rst [hunk 6]
```diff
@@ -0,0 +1,387 @@
...
+    sylius_resource:
+        resources:
+            app.shipping_method_image:
+                classes:
+                    model: AppBundle\Entity\ShippingMethodImage
+
+7. Create the ShippingMethodImageType class
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This is how the class for ``ShippingMethodImageType`` should look like. Place it in the ``AppBundle\Form\Type\ShippingMethod`` directory.
```
Comment (GSadee) [inline docs/cookbook/images-on-entity.rst]: Why not AppBundle\Form\Type directory?

Comment (TheMadeleine) [inline docs/cookbook/images-on-entity.rst]: Well... Don't know :<

**Comment by pjedrzejewski** [comment]
Nice, thanks Magda! :)

**Comment by pjedrzejewski** [comment]
@sweoggy @venyii Guys, thanks for the help with the docs - we need more eyes here! :)

**Pull request merged**